qualtran/bloqs/state_preparation/state_preparation_alias_sampling.py

```diff
@@ -235,21 +233,31 @@ def qrom_bloq(self) -> QROM:
             (self.alternates_bitsize, self.keep_bitsize),
         )
 
-    def decompose_from_registers(
-        self,
-        *,
-        context: cirq.DecompositionContext,
-        **quregs: NDArray[cirq.Qid],  # type:ignore[type-var]
-    ) -> Iterator[cirq.OP_TREE]:
-        yield PrepareUniformSuperposition(self.n_coeff).on(*quregs['selection'])
+    def build_composite_bloq(self, bb, **soqs):
+        soqs['selection'] = bb.add(
+            PrepareUniformSuperposition(self.n_coeff), target=soqs['selection']
+        )
         if self.mu == 0:
```

Comment:
After the fix in #1092 I think `mu` would never be `0`. So the decomp can be simplified by unpacking the registers in the method signature

Reply:
Should also probably assert `mu > 0` in post init

Reply:
done